packages/eslint-plugin/lib/rules/no-dynamic-i18n-messages.js

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+//------------------------------------------------------------------------------
+// Helpers
+//------------------------------------------------------------------------------
+
+const {isTextLabelChild, report} = require('../../util');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+/**
+ * @type {import('eslint').Rule.RuleModule}
+ */
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'enforce translate calls to be plain text labels',
+      category: 'Possible Problems',
+      url: 'https://github.com/facebook/docusaurus/tree/main/packages/eslint-plugin/docs/rules/no-dynamic-i18n-messages.md',
+    },
+    schema: [],
+    messages: {
+      translateChildren:
+        '<Translate> children must be hardcoded strings. You can have in-string dynamic placeholders using the values prop.',
+      translateArg:
+        'translation message must be a hardcoded string. You can have in-string dynamic placeholders using the values argument.',
+    },
+  },
+
+  create(context) {
+    const isMessageTypeValid = (type) => type === 'Literal';
+
+    const isNodeValid = (node) =>
+      node.children.every((child) => isTextLabelChild({child}));
```

Comment:
It would be useful to add an option `ignoreJSXAttributes`. We may want to flag `<div aria-label="untranslated" />` as well. Doesn't have to be in this PR though

Reply:
I didn't understand that, could you provide an example usage of `ignoreJSXAttributes` ? Also what is the purpose of `<div aria-label="untranslated" />` ? Self-closing tags do not have children (excluding the children prop) therefore the current rules do not apply in this case, right ?

Reply:
Oh, sorry, wrong comment place. I was talking about `no-untranslated-text`. There should be an option that turns on warning for JSX attributes like `aria-label` as well, not just string literal children.